### PR TITLE
Add blocking method for vApp status change tracking

### PR DESCRIPTION
**Pull request by Didainius**
Part of solution to https://github.com/terraform-providers/terraform-provider-vcd/issues/68
More details in the terraform-provider-vcd PR.

### CODE REVIEW

File: govcd/vapp.go [hunk 1]
```diff
@@ -567,6 +568,31 @@ func (vapp *VApp) GetStatus() (string, error) {
 	return types.VAppStatuses[vapp.VApp.Status], nil
 }
 
+// StatusWaitNot blocks until the status of vApp is not equal to unwantedStatus.
```
Comment (lvirbalas) [inline govcd/vapp.go]: I believe you meant "blocks until the status of vApp _is_ equal to unwantedStatus"

File: govcd/vapp.go [hunk 2]
```diff
@@ -567,6 +568,31 @@ func (vapp *VApp) GetStatus() (string, error) {
...
+// It sleeps 200 milliseconds between iterations and times out after timeOutAfterSeconds
+// of seconds.
+func (vapp *VApp) StatusWaitNot(unwantedStatus string, timeOutAfterSeconds int) error {
```
Comment (lvirbalas) [inline govcd/vapp.go]: Suggesting a different name for the function:
`BlockOnUnwantedStatus`

File: govcd/vapp_test.go [hunk 3]
```diff
@@ -106,6 +107,53 @@ func (vcd *TestVCD) Test_Reboot(check *C) {
...
+			t   Task
+			err error
+		}{task, err}
+	}()
+
+	// This must timeout as the timeout is zero
+	errMustTimeout := vcd.vapp.StatusWaitNot(initialVappStatus, 0)
+	check.Assert(errMustTimeout, ErrorMatches, "timed out waiting for vapp to become not .* seconds")
+
+	// This must wait until
```
Comment (lvirbalas) [inline govcd/vapp_test.go]: "This must wait until" ... until what? :)

File: govcd/vapp.go [hunk 4]
```diff
@@ -567,6 +568,31 @@ func (vapp *VApp) GetStatus() (string, error) {
...
+	timeoutAfter := time.After(time.Duration(timeOutAfterSeconds) * time.Second)
+	tick := time.Tick(200 * time.Millisecond)
+
+	for {
+		select {
+		case <-timeoutAfter:
+			return fmt.Errorf("timed out waiting for vapp to become not %s after %d seconds",
```
Comment (lvirbalas) [inline govcd/vapp.go]: vapp -> vApp

Comment (lvirbalas) [reply govcd/vapp.go]: Suggesting different wording:
"timed out waiting for vApp to _exit state_ %s after %d seconds"

**Comment by Didainius** [comment]
I can surely work more on documenting it better. Just to have a full picture I've added you as reviewers on PR in https://github.com/terraform-providers/terraform-provider-vcd/pull/189 for general view.

**Pull request merged**